src/test/unit/math/prim/mat/fun/value_of_rec_test.cpp

```diff
@@ -26,11 +26,17 @@ TEST(MathMatrix,value_of_rec) {
   
   Eigen::Matrix<double,2,5> a; 
   fill(a_vals, a);
+  Eigen::Map< Eigen::Matrix<double,2,5> > am(a.data(), a.rows(), a.cols());
```

Comment:
one question: is an `Eigen::Map< MatrixXd >` sufficient for testing `MatrixBase< Derived >`?
(the answers would either be 'yes' or 'no, we should also be testing ___'. I'm not sure if we need to add a couple tests for some of Eigen's expression templates (say `am.transpose()` or if `Eigen::Map` is sufficient)